cmd/operator/main.go

```diff
@@ -266,11 +267,28 @@ func Main() int {
 		return 1
 	}
 
-	pao, err := prometheusagentcontroller.New(ctx, cfg, log.With(logger, "component", "prometheusagentoperator"), r)
+	var pao *prometheusagentcontroller.Operator
+	verbs := map[string][]string{
+		monitoringv1alpha1.PrometheusAgentName:                           {"get", "list", "watch"},
+		fmt.Sprintf("%s/status", monitoringv1alpha1.PrometheusAgentName): {"update"},
+	}
+
+	cr := k8sutil.NewCRConfig(cfg.Namespaces.AllowList,
+		verbs,
+		monitoringv1alpha1.SchemeGroupVersion.String(),
+		monitoringv1alpha1.PrometheusAgentsKind,
+		monitoringv1alpha1.PrometheusAgentName)
+
+	err = cr.Runnable(ctx, cfg.Host, cfg.TLSInsecure, &cfg.TLSConfig)
 	if err != nil {
```

Comment:
we need to distinguish between 2 classes of error:
* the API server returned an unexpected error -> the operator needs to fail
* the prerequisites aren't met -> the controller can't be started.

I'd recommend wrapping a sentinel error (see https://bitfieldconsulting.com/golang/wrapping-errors) to detect the second case.